app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -293,7 +294,12 @@ open class FenixApplication : LocaleAwareApplication() {
                 components.core.store,
                 onNewTabOverride = {
                     _, engineSession, url ->
-                        val session = Session(url, components.browsingModeManager.mode.isPrivate)
+                        val shouldCreatePrivateSession =
```

Comment:
This is the alternative fix to messing with making the `browsingModeManager` a singleton on `components`. Ultimately we need to move towards that solution but there is a bit more we need to think about that we can do post code freeze